src/sentry/api/endpoints/project_performance_issue_settings.py

```diff
@@ -69,14 +65,21 @@ def put(self, request: Request, project) -> Response:
         if not serializer.is_valid():
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-        performance_issue_settings = projectoptions.get_well_known_default(
+        performance_issue_settings_default = projectoptions.get_well_known_default(
             SETTINGS_PROJECT_OPTION_KEY,
             project=project,
         )
 
+        performance_issue_settings = project.get_option(
+            SETTINGS_PROJECT_OPTION_KEY, default=performance_issue_settings_default
+        )
+
```

Comment:
Honestly shocking that this doesn't happen automatically under-the-hood! Seems wild to make you manually fetch the default ahead of time, given that options are registered with a default

Reply:
Well you get the default back, but in the case of dicts it doesn't merge them which is why I need to pull both. Bit of an edge case we'd probably be better off just having 1 option per detector to simplify things instead of an object.